builders/ethereum/dev-env/thirdweb.md

```diff
@@ -110,184 +112,180 @@ thirdweb offers SDKs for a range of programming languages, such as React, React
+});
 ```
 
+You can delete the placeholder code in `App.tsx` because we'll be replacing it as part of the tutorial. 
+
+### Configure Chain {: #configure-chain }
+
+thirdweb offers a small number of chains from `@thirdweb/chains` and does not include Moonbeam networks in that list, so you'll need to specify the network details including chain ID and RPC URL. You can create a custom chain with [`defineChain`](https://portal.thirdweb.com/references/typescript/v5/defineChain){target=\_blank} as follows:
+
+=== "Moonbeam"
+
+    ```typescript
+    import { defineChain } from 'thirdweb';
+    const moonbase = defineChain({
+    id: {{ networks.moonbeam.chain_id }},
+    rpc: '{{ networks.moonbeam.public_rpc_url }}',
+    })
+    ```
+
+=== "Moonriver"
+
+    ```typescript
+    import { defineChain } from 'thirdweb';
+    const moonbase = defineChain({
+    id: {{ networks.moonriver.chain_id }},
+    rpc: '{{ networks.moonriver.public_rpc_url }}',
+    })
+    ```
+
+=== "Moonbase Alpha"
+
+    ```typescript
+    import { defineChain } from 'thirdweb';
+    const moonbase = defineChain({
+    id: {{ networks.moonbase.chain_id }},
+    rpc: '{{ networks.moonbase.rpc_url }}',
+    })
+    ```
+
+
 ### Get Contract {: #get-contract }
 
-To connect to your contract, use the SDK’s [`getContract`](https://portal.thirdweb.com/typescript/sdk.thirdwebsdk.getcontract){target=\_blank} method.
+To connect to your contract, use the SDK’s [`getContract`](https://portal.thirdweb.com/references/typescript/v5/getContract){target=\_blank} method.  As an example, you can use Thirdweb to fetch data from an [Incrementer contract on Moonbase Alpha](https://moonbase.moonscan.io/address/0xa72f549a1a12b9b49f30a7f3aeb1f4e96389c5d8){target=\_blank}.
 
-```javascript
-import { useContract } from '@thirdweb-dev/react';
+```typescript title="App.tsx"
+    --8<-- 'code/builders/ethereum/dev-env/thirdweb/App.tsx:4:4'
+    --8<-- 'code/builders/ethereum/dev-env/thirdweb/App.tsx:6:6'
 
-function App() {
-  const { contract, isLoading, error } = useContract('INSERT_CONTRACT_ADDRESS');
-}
+    --8<-- 'code/builders/ethereum/dev-env/thirdweb/App.tsx:17:57'
 ```
 
 ### Calling Contract Functions {: #calling-contract-functions }
 
-For extension based functions, use the built-in supported hooks. There are several hooks available for you to use, the following are a few examples:
+To call a contract in the latest version of the SDK, you can use [`prepareContractCall`](https://portal.thirdweb.com/typescript/v5/transactions/prepare){target=\_blank}.
 
-- Use the NFTs extension to access a list of NFTs owned by an address via the [`useOwnedNFTs` hook](https://portal.thirdweb.com/react/react.useownednfts){target=\_blank}:
+```typescript title="App.tsx"
+    --8<-- 'code/builders/ethereum/dev-env/thirdweb/App.tsx:6:6'
 
-    ```javascript
-    import { useOwnedNFTs, useContract, useAddress } from '@thirdweb-dev/react';
+    --8<-- 'code/builders/ethereum/dev-env/thirdweb/App.tsx:144:148'
```

Comment:
This doesn't look right. I just looked into it and it looks like we can remove the indents, can you try adding the config to mkdocs.yml https://facelessuser.github.io/pymdown-extensions/extensions/snippets/#options

I guess we would need to set `dedent_subsections` to true